### DISCUSSION TITLE
Password fields and HTML5 compatability

### DISCUSSION THREAD

**Discussion by brittainmark** [zencart/zencart#5038] [General]
Good practice is for autocomplete to be set to off for password fields. 
HTML5 objects to autocomplete, being set to both "current-password" or "new-password" and "off". Which is what we do in zencart
However, most browsers allow the user to override this setting and save their passwords.

If we want to keep autocomplete set to "off" then we should remove "current-password" and "new-password" from all calls to zen_draw_password_field OR override autocomplete in zen_draw_password_field if it is set.

Thoughts?

**Comment by zcwilt** [comment]
Do we want autocomplete set to off completely. How does this affect e.g saved passwords coming from Firefox etc.
I guess to some degree I agree, in that for sites where I want security then I rely on a password safe(lastpass etc) to enter credentials and never save the password to to the browser.

Though generally that's for sites where I would have 2fa anyway, e.g aws console, server hosting sites etc. 

For Zen Cart I can see the advantage security wise for Admin login, but would we do the same for Customer login

**Reply by brittainmark** [reply]
Browser ignore autocomplete off if the user has set their browser to allow passwords to be stored. (I have tried with both Firefox and Edge, assume chrome will mirror edge)

**Comment by dbltoe** [comment]
Notice that they have set it to _Won't Fix_.
https://bugs.chromium.org/p/chromium/issues/detail?id=468153
